skopt/utils.py

```diff
@@ -501,3 +499,104 @@ def normalize_dimensions(dimensions):
+    the objective function: `func(x=[123, 3.0, 'hello'])`. This wrapper
+    converts your objective function with named arguments into one that
+    accepts a list as argument, while doing the conversion automatically.
+
+    The advantage of this is that you don't have to unpack the list of
+    arguments `x` yourself, which makes the code easier to read and
+    also reduces the risk of bugs if you change the number of dimensions
+    or their order in the search-space.
+
+    Example Usage
+    -------------
+    # Define the search-space dimensions. They must all have names!
+    dim1 = Real(name='foo', low=0.0, high=1.0)
+    dim2 = Real(name='bar', low=0.0, high=1.0)
+    dim3 = Real(name='baz', low=0.0, high=1.0)
+
+    # Gather the search-space dimensions in a list.
+    dimensions = [dim1, dim2, dim3]
+
+    # Define the objective function with named arguments
+    # and use this function-decorator to specify the search-space dimensions.
+    @use_named_args(dimensions=dimensions)
+    def my_objective_function(foo, bar, baz):
+        return foo ** 2 + bar ** 4 + baz ** 8
+
+    # Now the function is callable from the outside as
+    # `my_objective_function(x)` where `x` is a list of unnamed arguments,
+    # which then wraps your objective function that is callable as
+    # `my_objective_function(foo, bar, baz)`.
+    # The conversion from a list `x` to named parameters `foo`, `bar`, `baz`
+    # is done automatically.
+
+    # Run the optimizer on the wrapped objective function which is called as
+    # `my_objective_function(x)` as expected by `forest_minimize()`.
+    result = forest_minimize(func=my_objective_function, dimensions=dimensions,
+                             n_calls=20, base_estimator="ET", random_state=4)
+
+    # Print the best-found results.
+    print("Best fitness:", result.fun)
+    print("Best parameters:", result.x)
+
+    Parameters
+    ----------
+    * `dimensions` [list(Dimension)]:
+        List of `Dimension`-objects for the search-space dimensions.
+
+    Returns
+    -------
+    * `wrapped_func` [callable] 
+        Wrapped objective function.
+    """
+
+    def decorator(func):
+        """
+        This uses more advanced Python features to wrap `func` using a
+        function-decorator. To get explanations of how this works in Python,
+        please do an internet search for e.g.:
+        Python function decorator with additional arguments.
+        """
+
+        # Ensure all dimensions are correctly typed.
+        if not all(isinstance(dim, Dimension) for dim in dimensions):
+            raise ValueError("All dimensions must be instances of the Dimension-class.")
+
+        # Ensure all dimensions have names.
+        if any(dim.name is None for dim in dimensions):
+            raise ValueError("All dimensions must have names.")
+
+        @wraps(func)
+        def wrapper(x):
+            # Ensure the number of dimensions match
```

Comment:
Consider adding a docstring explaining the type of x and what it is. It seems that Pycharm for instance can infer expected type of variable from docstring, providing extra variable checks and warnings, which sometimes save me bugs. Possibly could also simplify comments as per discussion below

Reply:
Done.